.github/workflows/duplicate_pr.yml

```diff
@@ -0,0 +1,77 @@
+name: Duplicate PR
+
+#Duplication workflow must be run before PR branch is deleted
+on:
+  issue_comment:
+    types: [created, edited]
+
+jobs:
+  duplicate_pr:
+    name: Duplicate PR
+    if: ${{ github.event.issue.pull_request && startsWith(github.event.comment.body,'@dimagibot duplicate this PR') }}
+    runs-on: ubuntu-latest
+    steps:
+      # https://github.com/marketplace/actions/github-script
+      - name: Get PR Intial Commit SHA
+        if: ${{ github.event.issue.pull_request.merged_at }}
+        id: initial-sha
+        uses: actions/github-script@0.9.0
+        with:
+          script: |
+            const sha = context.payload.comment.body.split(" ").pop()
+            core.setOutput('sha', sha)
+      # https://github.com/marketplace/actions/pull-request-comment-branch
+      - name: Pull Request Comment Branch
+        uses: xt0rted/pull-request-comment-branch@v2.0.0
+        id: comment-branch
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          fetch-depth: 0
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+      - name: Install Requirements
+        run: pip install sh
+      - name: Config Git
+        run: |
+          git config --global user.email github-actions@github.com
+          git config --global user.name github-actions
+      - name: Run duplicate_pr.py script with PR merged
+        if: ${{ github.event.issue.pull_request.merged_at }}
+        run: |
+          echo Intial SHA is "${{ steps.initial-sha.outputs.sha }}"
+          python scripts/duplicate_pr.py ${{ steps.comment-branch.outputs.head_ref }} ${{ steps.comment-branch.outputs.base_ref }} --initial_sha ${initial_sha}
+        env:
+          initial_sha: ${{ steps.initial-sha.outputs.sha }}
+      - name: Run duplicate_pr.py script with PR not merged
```

Comment:
curious why do we need to do things differently depending on if PR is merged or not ?

Reply:
The script uses `git merge-base` to find the commit to start cherrypicking frame. What `git merge-base` does is find the last common ancestor between two branches. So if the commits are
Formplayer: A>B>C
New Branch: A>B>C>D>E
it would return commit C. But once new branch gets merged into Formplayer, there isn't a way to distinguish where the diverging commit and the last common ancestor would be commit E.

I wasn't able to find a paylod from GitHub action that provide the "first" commit of a PR or any other way to get this info.

Reply:
Makes sense, thanks!